packages/bootleg-app-contracts/migrations/2_deploy_and_mint.js

```diff
@@ -35,6 +41,14 @@ module.exports = async (deployer, network, accounts) => {
   // Set the token to have the app as the only trader
   await token.setTraderApp(app.address);
 
+  // The upload the json metadata defined by ERC721
+  const path = './token_1_metadata.json';
+  const ipfs = ipfsClient(ipfsConfig.host, ipfsConfig.port, ipfsConfig.options);
+  const ipfsResult = await ipfs.addFromFs(path, {});
```

Comment:
You should add `{pin: true}` as the options. Otherwise, infura might garbage collect the file.

Reply:
Yeah, I saw that in your code but did not see it as an option here: https://github.com/ipfs/interface-js-ipfs-core/blob/master/SPEC/FILES.md#addfromfs
Also, and if they use the `add` under-the-hood in `addfromfs` it seems the default is `true` https://github.com/ipfs/interface-js-ipfs-core/blob/master/SPEC/FILES.md#add
Maybe it wasn't always the default?

Reply:
I uh... I can't figure out if a file is pinned or not by default. There also is, as far as I can tell, no way to know if ipfs knows that the file is pinned. I'll just add `{pin: true}` because I can't spend any more time on this.